src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SymbolsTests.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using UnifiedBuildTasks = Microsoft.DotNet.UnifiedBuild.Tasks;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+public class SymbolsTests : SdkTests
+{
+    private static string SymbolsTestsRoot { get; } = Path.Combine(Directory.GetCurrentDirectory(), nameof(SymbolsTests));
+
+    public SymbolsTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+    /// <summary>
+    /// Verifies that all symbols have valid sourcelinks.
+    /// </summary>
+    [Fact]
+    public void VerifySdkSymbols()
+    {
+        try
+        {
+            if (Directory.Exists(SymbolsTestsRoot))
+            {
+                Directory.Delete(SymbolsTestsRoot, true);
+            }
+            Directory.CreateDirectory(SymbolsTestsRoot);
+
+            string symbolsRoot = Directory.CreateDirectory(Path.Combine(SymbolsTestsRoot, "symbols")).FullName;
+
+            // We are validating dotnet-symbols-sdk-*.tar.gz which contains source-built sdk symbols
+            Utilities.ExtractTarball(
+                Utilities.GetFile(Path.GetDirectoryName(Config.SourceBuiltArtifactsPath)!, "dotnet-symbols-sdk-*.tar.gz"),
+                symbolsRoot,
+                OutputHelper);
+
+            string sdkRoot = Directory.CreateDirectory(Path.Combine(SymbolsTestsRoot, "sdk")).FullName;
+            Utilities.ExtractTarball(Config.SdkTarballPath!, sdkRoot, OutputHelper);
```

Comment:
This already exists at Config.DotNetDirectory once DotNetHelper has been instantiated.  Consider using it to avoid extracting this tarball again.

Reply:
Fixed with https://github.com/dotnet/installer/pull/19528/commits/e431f1157fb33cb7977252e5ad9567ae53316a45